### PR TITLE
Add total pages to MendeleyScraper

### LINKED ISSUE #25: MendeleyScraper class doesn't implement total_pages which results total = "Unknown"
`MendeleyScraper` should implement `total_pages` method, as the current implementation uses the `ScraperBase`'s `total_pages` which returns: `Unknown` which breaks the articles importing flow.

### PR DESCRIPTION
**Pull request by kohrVid**
Closes #25

The Mendeley Support team have confirmed that the `Mendeley-Count` header is only available in partial responses. This means that it won't appear on the response of the first page of a collection when the collection length is lower than or equal to the value of the `MENDELEY_RESULTS_PER_PAGE` environment variable.  As such, the `response_length` must be counted to determine the totals for smaller collections.

Note, when multiple collections are scraped at once, a combination of "Unknown" and exact totals may appear in the MendeleyScraper's output.

### CODE REVIEW

File: lib/rayyan-scrapers/mendeley_scraper.rb [hunk 1]
```diff
@@ -146,6 +146,24 @@ def parse_search_results(page, extraction_fields = nil)
...
 
+    def total_pages(_)
+      url = create_search_url
+      page = Typhoeus::Request.get(url, @headers)
+      handle_error_response(page)
+      response_length = parse_page(page).length
+
+      total = if page.is_a?(Typhoeus::Response) &&
+                    page.headers['Mendeley-Count'].present?
+                page.headers['Mendeley-Count']
```
Comment (hammady) [inline lib/rayyan-scrapers/mendeley_scraper.rb:157]: Try converting to an integer.

File: lib/rayyan-scrapers/mendeley_scraper.rb [hunk 2]
```diff
@@ -146,6 +146,24 @@ def parse_search_results(page, extraction_fields = nil)
...
+              elsif response_length <= self.class.results_per_page
+                response_length.to_s
+              else
+                'Unknown'
+              end
+
+      total
```
Comment (hammady) [inline lib/rayyan-scrapers/mendeley_scraper.rb:164]: No need to set `total` variable and return it.

File: lib/rayyan-scrapers/mendeley_scraper.rb [hunk 3]
```diff
@@ -146,6 +146,24 @@ def parse_search_results(page, extraction_fields = nil)
...
+      handle_error_response(page)
+      response_length = parse_page(page).length
+
+      total = if page.is_a?(Typhoeus::Response) &&
+                    page.headers['Mendeley-Count'].present?
+                page.headers['Mendeley-Count']
+              elsif response_length <= self.class.results_per_page
+                response_length.to_s
+              else
+                'Unknown'
```
Comment (hammady) [inline lib/rayyan-scrapers/mendeley_scraper.rb:161]: Delegate to `super` in this case.

**Review by hammady — CHANGES_REQUESTED** [review]

File: lib/rayyan-scrapers/mendeley_scraper.rb (hunk 1 above)
Comment (kohrVid) [reply lib/rayyan-scrapers/mendeley_scraper.rb]: No problem, I've updated the PR so that the method returns an integer going forwards

**Review by hammady — APPROVED** [review]

**Pull request merged**